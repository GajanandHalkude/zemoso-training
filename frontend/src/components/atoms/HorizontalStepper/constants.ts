export const steps = [
  'Amount',
  'You',
  'Recipient',
  'Verification',
  'Review',
  'Pay'
]
