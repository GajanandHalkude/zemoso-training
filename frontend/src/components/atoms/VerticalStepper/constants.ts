export const steps = [
  { time: 'Today at 6:43 pm', label: 'You set up your transfer' },
  { time: 'Today at 6:44 pm', label: 'We recieved your GBP' },
  { time: 'Today at 6:50 pm', label: 'Your moneyâ€™s being processed' },
  { time: 'Tomorrow at 12:00 am', label: 'We pay out your EUR' },
  { time: 'Tomorrow at 6:00 am', label: 'George max recieves your EUR' }
]
