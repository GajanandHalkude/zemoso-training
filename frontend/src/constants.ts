import Bitcoin  from "../public/assets/images/bitcoin.svg";
import Ethreum  from "../public/assets/images/ethreum.svg";
import Xrp  from "../public/assets/images/Rectangle 4.svg";
import Tether  from "../public/assets/images/Tether.svg";
import BitcoinCoin from "../public/assets/images/Bitcoin Coin.svg";
import Cardano from "../public/assets/images/Cardano.svg";
import DodgeCoin from "../public/assets/images/Dodge Coin.svg";
import Ethereum2 from "../public/assets/images/ethereum 2.svg";
import xrp from "../public/assets/images/XRP.svg";
import USDCoin from "../public/assets/images/USD Coin.svg"
import theme from '../src/theme'
import Google from "../public/assets/icons/google.svg";
import Facebook from "../public/assets/icons/facebook.svg";
import Microsoft from "../public/assets/icons/microsoft.svg";
import ButtonComponent from "../src/components/atoms/button";
import CustomTextField from "../src/components/atoms/textField";
import MuiTypography from "../src/components/atoms/typography";
import SocialLogin from "../src/components/molecules/socialLogin";
import moment from 'moment'


export { ButtonComponent, CustomTextField, MuiTypography, SocialLogin };

 export const socialLoginOptions = [
  { src: Google, text: "Google" },
  { src: Facebook, text: "Facebook" },
  { src: Microsoft, text: "Microsoft" },
];


export const menuItems = [
    { text: "Dashboard" },
    { text: "Careers" },
    { text: "Legal & Privacy" },
    { text: "© 2021 Minet", isBlack: true },
  ];

  export const buttonLabel = "NEED HELP";

  export const CHOOSE_CRYPTO = "Choose crypto";

  export const currencyBannerContent = "The world’s first cryptocurrency, Bitcoin is stored and exchanged securely on the internet through a digital ledger known as a blockchain. Bitcoins are divisible into smaller units known as satoshis each satoshi is worth 0.00000001 bitcoin."
  
  export const currencyBannerData = [
    {
        icon:Bitcoin,coinName:"Bitcoin" , shortName:"Moves tightly together" , value:3285553.375 , percentage:100
    },
    {
      icon:Ethreum,coinName:"Ethreum" , shortName:"Moves tightly together" , value:230966.85 , percentage:86
    },
    {
      icon:Xrp,coinName:"XRP" , shortName:"Moves tightly together" , value:60.20 , percentage:10
    },
    {
      icon:Tether,coinName:"Tether" , shortName:"Moves tightly together" , value:74.28 , percentage:2
    },
  ]
   export const  buy="BUY";
   export const sell="SELL";
export const formatCurrency = new Intl.NumberFormat('en-US', {
  style: 'currency',
  currency: 'USD',
})

export const paymentSummaryStepper = {
  step1: ["Payment method", "Visa credit ...8845"],
  step2: ["Delivery fees", "0.001 BTC"],
  step3: ["Deposit to", "Bitcoin wallet"],
};

export const dollarValue = 3406069.54;

export const timeLineValues = ["1H", "24H", "1W", "1M", "1Y", "ALL"];   

export const cryptoCoins = [
  {
    name: 'Bitcoin',
    color: theme.palette.chipColors.main,
  },
  {
    name: 'XRP',
    color: theme.palette.chipColors.color1,
  },
  {
    name: 'Polkadot',
    color: theme.palette.chipColors.color2,
  },
  {
    name: 'Ethereum',
    color: theme.palette.chipColors.color3,
  },
  {
    name: 'Tether',
    color: theme.palette.chipColors.color4,
  },
  {
    name: 'Ethereum 2',
    color: theme.palette.chipColors.color5,
  },
  {
    name: 'Dodge coin',
    color: theme.palette.chipColors.color6,
  },
]
export const passwordspecification = "A min of 8 character with atleast 1 special character and number included"

export const MyPortfolioData = [
  {
      icon:Bitcoin,coinName:"Bitcoin" , shortName:"BTC" , value:1.00 , percentage:0.00
  },
  {
    icon:Ethreum,coinName:"Ethreum" , shortName:"ETH" , value:2.00 , percentage:0.00
  },
  {
    icon:Tether,coinName:"Tether" , shortName:"USDT" , value:3.00 , percentage:0.00
  },
  {
    icon:Xrp,coinName:"XRP" , shortName:"XRP" , value:4.00 , percentage:0.00
  }
]


 export const ProfitData = [30, 40, 45, 80, 49, 60, 30, 91];
export const LossData = [50, 30, 40, 60, 40, 30, 90, 30];



export const WalletBodyData = [
  {
    date:new Date("2023-06-15") ,coinName:"Bitcoin" , TraderName:"Saiprabhu" , Trade:"Purchased" , QuantityOfCoin:0.001, Amount:9000
  },
  {
    date:new Date("2023-06-15") ,coinName:"Ethereum" , TraderName:"Gajanand" , Trade:"Sold" , QuantityOfCoin:0.002, Amount:7000
  }

]
interface HashmapProps {
  [key: string]: string;
}

export const pictures: HashmapProps = {
  bitcoin: Bitcoin,
  ethereum: Ethreum,
  usdCoin: USDCoin,
  xrp: xrp,
  binanceUsd: BitcoinCoin,
  ethereum2: Ethereum2,
  dogecoin: DodgeCoin,
  cardano: Cardano,
  tether: Tether,
};

export const currencies = [
  {
    id: "bitcoin",
    name: "Bitcoin",
    symbol: "BTC",
    icon: "bitcoin",
    price: 19124.56,
    marketCap: 366730983,
    totalSupply: 210000,
    availableSupply: 1917711,
    priceChangeIn1hr: -0.161,
    priceChangeIn24hrs: -0.51,
    priceChangeIn1week: -0.51,
    priceChangeIn1month: 2.51,
    priceChangeIn1year: 2.671,
    volume: 56214521561,
    isSelected: true,
  },
  {
    id: "ethereum",
    name: "Ethereum",
    symbol: "ETH",
    icon: "ethereum",
    price: 19123.561,
    marketCap: 366730983872,
    totalSupply: 210000002,
    availableSupply: 19177112,
    priceChangeIn1hr: -0.162,
    priceChangeIn24hrs: -0.52,
    priceChangeIn1week: -0.52,
    priceChangeIn1month: 2.52,
    priceChangeIn1year: 2.672,
    volume: 562145215,
    isSelected: false,
  },
  {
    id: "usd-coin",
    name: "USD Coin",
    symbol: "USDC",
    icon: "binanceUsd",
    price: 29123.56,
    marketCap: 6730983878,
    totalSupply: 31000000,
    availableSupply: 18177118,
    priceChangeIn1hr: -1.16,
    priceChangeIn24hrs: -1.5,
    priceChangeIn1week: -1.5,
    priceChangeIn1month: 1.5,
    priceChangeIn1year: 1.67,
    volume: 564521562,
    isSelected: false,
  },
  {
    id: "ripple",
    name: "XRP",
    symbol: "XRP",
    icon: "xrp",
    price: 28223.56,
    marketCap: 376730983878,
    totalSupply: 29000000,
    availableSupply: 24177118,
    priceChangeIn1hr: -8.16,
    priceChangeIn24hrs: -4.5,
    priceChangeIn1week: -4.5,
    priceChangeIn1month: 24.5,
    priceChangeIn1year: 24.67,
    volume: 56214531562,
    isSelected: false,
  },
  {
    id: "ethereum2",
    name: "Ethereum 2",
    symbol: "ETH",
    icon: "ethereum2",
    price: 19123.56,
    marketCap: 366730983878,
    totalSupply: 21000000,
    availableSupply: 19177118,
    priceChangeIn1hr: -0.16,
    priceChangeIn24hrs: -0.5,
    priceChangeIn1week: -0.5,
    priceChangeIn1month: 2.5,
    priceChangeIn1year: 2.67,
    volume: 56214521562,
    isSelected: false,
  },
  {
    id: "cardano",
    name: "Cardano",
    symbol: "CRD",
    icon: "cardano",
    price: 89123.56,
    marketCap: 466930183878,
    totalSupply: 410000,
    availableSupply: 191937118,
    priceChangeIn1hr: -30.16,
    priceChangeIn24hrs: -30.5,
    priceChangeIn1week: -30.5,
    priceChangeIn1month: 26.5,
    priceChangeIn1year: 26.67,
    volume: 56214575462,
    isSelected: false,
  },
  {
    id: "dodge",
    name: "Dodge Coin",
    symbol: "DDG",
    icon: "dogecoin",
    price: 53623.56,
    marketCap: 3789630983878,
    totalSupply:14592000,
    availableSupply: 1975410118,
    priceChangeIn1hr: -7.16,
    priceChangeIn24hrs: -7.5,
    priceChangeIn1week: -7.5,
    priceChangeIn1month: 7.5,
    priceChangeIn1year: 7.67,
    volume: 56263821562,
    isSelected: false,
  },
  {
    id: "tether",
    name: "Tether",
    symbol: "ETH",
    icon: "tether",
    price: 45393.56,
    marketCap: 743690983878,
    totalSupply: 76150000,
    availableSupply: 95417118,
    priceChangeIn1hr: -6.16,
    priceChangeIn24hrs: -6.5,
    priceChangeIn1week: -6.5,
    priceChangeIn1month: 6.5,
    priceChangeIn1year: 6.67,
    volume: 5621457429,
    isSelected: false,
  },
];

export const compactFormat = new Intl.NumberFormat('en-US', {
  notation: 'compact',
  maximumFractionDigits: 1,
})
export const formatDate = (date: string) => {
  return moment(date, 'YYYY-MM-DD').format('LL').split(',')[0]
}
export const RECENT_TRANSACTIONS = "Recent Transactions"
export const VIEW_ALL = "View All"

export const isEmailValid = (email: string): boolean => {
  const emailRegex = /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/
  return emailRegex.test(email);
};

export const isPasswordValid = (password: string): boolean => {
  const regex = /^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!%*#?&]{8,}$/;
  return regex.test(password);
};

