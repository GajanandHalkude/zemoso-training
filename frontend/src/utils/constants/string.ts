export const SELECT_ACCOUNT_TYPE_HEADING1 =
  'What kind of account would you like to open today?'
export const SELECT_ACCOUNT_TYPE_HEADING2 =
  'You can add another account later on, too.'
export const SELECT_COUNTRY_HEADING = 'Your country of registration'
export const SELECT_COUNTRY_INPUT_PLACEHOLDER = 'Select your country'
export const ADD_NEW_ADDRESS_MODAL_HEADING = 'Add new address'
export const ADD_NEW_ADDRESS_MODAL_BUTTON_TITLE = 'Add'
