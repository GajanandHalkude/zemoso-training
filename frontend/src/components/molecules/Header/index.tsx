import { Box, Divider, Grid, styled } from '@mui/material'
import React from 'react'
import { NavLink } from 'react-router-dom'
import theme from '../../../theme'
import ButtonComponent from '../../atoms/button'
import MuiTypography from '../../atoms/typography'
import AvatarComponent from '../../atoms/avatar'
import IconComponent from '../../atoms/icon'
import chevronDropdown from "../../../../public/assets/images/avatardropdown.svg"
import {buy} from "../../../../src/constants"
import {sell} from "../../../../src/constants"


 export interface IHeaderProps {
    pageName: string;
    displayButtons?:boolean;
  }



const CustomBox = styled(Box)({
padding: '20px 24px 20px 24px',
[theme.breakpoints.up('lg')]: {
maxWidth: '1500px'
},
[theme.breakpoints.up('xl')]: {
maxWidth: '98vw'
},
[theme.breakpoints.down('lg')]: {
minWidth: '1366px'
},
[theme.breakpoints.between('lg', 'xl')]: {
width: '97vw',
},
borderBottom: `1px solid ${theme.palette.greyColors.grey100}`,
})

const CustomButton = styled(ButtonComponent)({
borderRadius: '4px',
padding: '0px 16px 0px 16px',
width: '120px',
height: '42px',
})

const StyledIconContainer = styled(Grid)({
    paddingTop: '7px',
})

const Header: React.FC<IHeaderProps> = ({pageName, displayButtons}:IHeaderProps) => {

return (
<CustomBox data-testid="Header">
<Grid container>
<Grid item xs={6}>
<MuiTypography
         variant="h6"
         width="122px"
         height="34px"
         color="textColor.highEmphasis"
         text={pageName}
       />
</Grid>

    <Grid item container wrap="nowrap" xs={6} justifyContent="flex-end" columnGap={2}>
        {displayButtons && 
        <Grid  display="flex" alignItems="row" gap="12px">
      <Grid item>
        <NavLink to="/sell" style={{ textDecoration: 'none' }}>
          <CustomButton variant="contained" backgroundColor="#FFA74F" text={sell}/>
        </NavLink>
      </Grid>

      <Grid item>
        <NavLink to="/purchase" style={{ textDecoration: 'none' }}>
          <CustomButton variant="contained" backgroundColor="#0052FF" text={buy} />
        </NavLink>
      </Grid>
      </Grid>
}
      <Divider orientation="vertical" variant="middle" flexItem />
      <Grid item display="flex" alignItems="center"  gap="10px">
        <AvatarComponent height="30px" width="30px" />
        <StyledIconContainer item>
                <IconComponent width='13px' height='13px' src={chevronDropdown}/>
            </StyledIconContainer>
      </Grid>


    </Grid>
  </Grid>
</CustomBox>
)
}

export default Header