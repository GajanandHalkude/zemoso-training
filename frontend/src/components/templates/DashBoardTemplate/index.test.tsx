/* eslint-disable react/no-children-prop */
import renderer from 'react-test-renderer';
import DashBoardTemplate from "./index";
import {render, screen} from '@testing-library/react';
import '@testing-library/jest-dom'
import { BrowserRouter } from 'react-router-dom';
import Header from '../../molecules/Header';
import Footer from '../../molecules/footer';
import SideNavCompnent from '../../molecules/sideNavbar';
import { menuItems } from '../../../constants';
import React from 'react';

it('renders home template correctly', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <DashBoardTemplate
            header={<Header pageName="Dashboard" displayButtons />}
            footer={<Footer menuItems={menuItems} buttonLabel="Need Help" />}
            children={
              <div style={{ height: "896px", width: "1296px" }}>content</div>
            }
            sideNav={<SideNavCompnent />}
            paddingTop="10px"
          />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
});

describe("Templates/DashBoard Template", () => {
  test("Renders home template correctly", () => {
    render(
      <BrowserRouter>
        <DashBoardTemplate
          header={<Header pageName="Dashboard" displayButtons />}
          footer={<Footer menuItems={menuItems} buttonLabel="Need Help" />}
          children={
            <div style={{ height: "896px", width: "1296px" }}>content</div>
          }
          sideNav={<SideNavCompnent />}
        />
      </BrowserRouter>
    );
    const template = screen.getByTestId("home-template");
    expect(template).toBeInTheDocument();
  });
});