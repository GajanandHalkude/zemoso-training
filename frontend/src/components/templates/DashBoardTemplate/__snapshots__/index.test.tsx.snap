// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders home template correctly 1`] = `
<div
  className="MuiBox-root css-0"
  data-testid="home-template"
>
  <div
    className="MuiBox-root css-waw6ev"
  >
    <div
      className="MuiBox-root css-1ggoh19"
      data-testid="sideNav"
    >
      <img
        data-testid="image"
        height="33px"
        src=""
        width="32px"
      />
      <div
        className="MuiBox-root css-l9wl2g"
      >
        <ul
          className="MuiList-root MuiList-padding css-h4y409-MuiList-root"
        >
          <li
            className="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
            disabled={false}
            style={
              {
                "margin": "14px",
              }
            }
          >
            <a
              className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters list-item-button css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
              href="/dashboard"
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
            >
              <div
                aria-label="Dashboard"
                aria-labelledby={null}
                className="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
                data-mui-internal-clone-element={true}
                onBlur={[Function]}
                onFocus={[Function]}
                onMouseLeave={[Function]}
                onMouseOver={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
              >
                <div>
                  <img
                    alt="icon"
                    data-testid="iconComponent"
                    height="32px"
                    src=""
                    style={
                      {
                        "padding": undefined,
                      }
                    }
                    width="32px"
                  />
                </div>
              </div>
              <span
                className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </a>
          </li>
          <li
            className="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
            disabled={false}
            style={
              {
                "margin": "14px",
              }
            }
          >
            <div
              className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters list-item-button css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              <div
                aria-label="My Portfolio"
                aria-labelledby={null}
                className="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
                data-mui-internal-clone-element={true}
                onBlur={[Function]}
                onFocus={[Function]}
                onMouseLeave={[Function]}
                onMouseOver={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
              >
                <div>
                  <img
                    alt="icon"
                    data-testid="iconComponent"
                    height="32px"
                    src=""
                    style={
                      {
                        "padding": undefined,
                      }
                    }
                    width="32px"
                  />
                </div>
              </div>
              <span
                className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </div>
          </li>
          <li
            className="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
            disabled={false}
            style={
              {
                "margin": "14px",
              }
            }
          >
            <div
              className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters list-item-button css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              <div
                aria-label="My Portfolio"
                aria-labelledby={null}
                className="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
                data-mui-internal-clone-element={true}
                onBlur={[Function]}
                onFocus={[Function]}
                onMouseLeave={[Function]}
                onMouseOver={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
              >
                <div>
                  <img
                    alt="icon"
                    data-testid="iconComponent"
                    height="32px"
                    src=""
                    style={
                      {
                        "padding": undefined,
                      }
                    }
                    width="32px"
                  />
                </div>
              </div>
              <span
                className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </div>
          </li>
          <li
            className="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
            disabled={false}
            style={
              {
                "margin": "14px",
              }
            }
          >
            <div
              className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters list-item-button css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              <div
                aria-label="My Portfolio"
                aria-labelledby={null}
                className="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
                data-mui-internal-clone-element={true}
                onBlur={[Function]}
                onFocus={[Function]}
                onMouseLeave={[Function]}
                onMouseOver={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
              >
                <div>
                  <img
                    alt="icon"
                    data-testid="iconComponent"
                    height="32px"
                    src=""
                    style={
                      {
                        "padding": undefined,
                      }
                    }
                    width="32px"
                  />
                </div>
              </div>
              <span
                className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </div>
          </li>
          <li
            className="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
            disabled={false}
            style={
              {
                "margin": "14px",
              }
            }
          >
            <a
              className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters list-item-button css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
              href="/signin"
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
            >
              <div
                aria-label="LogOut"
                aria-labelledby={null}
                className="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
                data-mui-internal-clone-element={true}
                onBlur={[Function]}
                onFocus={[Function]}
                onMouseLeave={[Function]}
                onMouseOver={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
              >
                <div>
                  <img
                    alt="icon"
                    data-testid="iconComponent"
                    height="32px"
                    src=""
                    style={
                      {
                        "padding": undefined,
                      }
                    }
                    width="32px"
                  />
                </div>
              </div>
              <span
                className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-6r5xil-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item css-4rmxrd-MuiGrid-root"
    >
      <div
        className="MuiBox-root css-914xdy"
        data-testid="Header"
      >
        <div
          className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
          >
            <h6
              className="MuiTypography-root MuiTypography-h6 css-dr70im-MuiTypography-root"
              data-testid="typography-component"
            >
              Dashboard
            </h6>
          </div>
          <div
            className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-wrap-xs-nowrap MuiGrid-grid-xs-6 css-19s4pri-MuiGrid-root"
          >
            <div
              className="MuiGrid-root css-1995vhn-MuiGrid-root"
            >
              <div
                className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <button
                  className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-ropxi6-MuiButtonBase-root-MuiButton-root"
                  data-testid="button"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  SELL
                  <span
                    className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
              <div
                className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <button
                  className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-4v2cgg-MuiButtonBase-root-MuiButton-root"
                  data-testid="button"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  BUY
                  <span
                    className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
            <hr
              className="MuiDivider-root MuiDivider-middle MuiDivider-vertical MuiDivider-flexItem css-16bghm4-MuiDivider-root"
            />
            <div
              className="MuiGrid-root MuiGrid-item css-19ysiih-MuiGrid-root"
            >
              <div
                className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-1jl820a-MuiAvatar-root"
                data-testid="avatar"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
                  data-testid="PersonIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                  />
                </svg>
              </div>
              <div
                className="MuiGrid-root MuiGrid-item css-hhlda7-MuiGrid-root"
              >
                <div>
                  <img
                    alt="icon"
                    data-testid="iconComponent"
                    height="13px"
                    src=""
                    style={
                      {
                        "padding": undefined,
                      }
                    }
                    width="13px"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item css-b6d1k6-MuiGrid-root"
    >
      <div
        style={
          {
            "height": "896px",
            "width": "1296px",
          }
        }
      >
        content
      </div>
    </div>
    <div
      className="MuiBox-root css-0"
      data-testid="footer"
    >
      <div
        className="MuiGrid-root MuiGrid-container css-vc2gs6-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
        >
          <div
            className="MuiStack-root css-u10pvi-MuiStack-root"
          >
            <p
              className="MuiTypography-root MuiTypography-body2 css-1eyzb45-MuiTypography-root"
              data-testid="typography-component"
            >
              Dashboard
            </p>
            <p
              className="MuiTypography-root MuiTypography-body2 css-1eyzb45-MuiTypography-root"
              data-testid="typography-component"
            >
              Careers
            </p>
            <p
              className="MuiTypography-root MuiTypography-body2 css-1eyzb45-MuiTypography-root"
              data-testid="typography-component"
            >
              Legal & Privacy
            </p>
            <p
              className="MuiTypography-root MuiTypography-body2 css-12fjmx1-MuiTypography-root"
              data-testid="typography-component"
            >
              © 2021 Minet
            </p>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-k97rmi-MuiGrid-root"
        >
          <div
            className="MuiStack-root css-149yd00-MuiStack-root"
          >
            <img
              data-testid="image"
              height="150px"
              src=""
              width="150px"
            />
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-o9t3yf-MuiButtonBase-root-MuiButton-root"
              data-testid="button"
              disabled={false}
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              Need Help
              <span
                className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
